Menus/TitleScreen.cpp

```diff
@@ -0,0 +1,608 @@
+				m_IntroSequenceState = IntroSequence::DataRealmsLogoFadeOut;
+			} else if (m_IntroSequenceState == IntroSequence::DataRealmsLogoFadeOut) {
+				m_IntroSequenceState = IntroSequence::FmodLogoFadeIn;
+			} else if (m_IntroSequenceState < IntroSequence::FmodLogoFadeOut) {
+				m_IntroSequenceState = IntroSequence::FmodLogoFadeOut;
+			}
+			return;
+		}
+
+		switch (m_IntroSequenceState) {
+			case IntroSequence::DataRealmsLogoFadeIn:
+				if (m_SectionSwitch) {
+					SetSectionDurationAndResetSwitch(0.5F);
+					g_GUISound.SplashSound()->Play();
+				}
+				m_FadeAmount = static_cast<int>(LERP(0, 1.0F, 255.0F, 0, m_SectionProgress));
+				break;
+			case IntroSequence::DataRealmsLogoDisplay:
+				if (m_SectionSwitch) { SetSectionDurationAndResetSwitch(2.0F); }
+				break;
+			case IntroSequence::DataRealmsLogoFadeOut:
+				if (m_SectionSwitch) { SetSectionDurationAndResetSwitch(0.25F); }
+				m_FadeAmount = static_cast<int>(LERP(0, 1.0F, 0, 255.0F, m_SectionProgress));
+				break;
+			case IntroSequence::FmodLogoFadeIn:
+				if (m_SectionSwitch) { SetSectionDurationAndResetSwitch(0.25F); }
+				m_FadeAmount = static_cast<int>(LERP(0, 1.0F, 255.0F, 0, m_SectionProgress));
+				break;
+			case IntroSequence::FmodLogoDisplay:
+				if (m_SectionSwitch) { SetSectionDurationAndResetSwitch(2.0F); }
+				break;
+			case IntroSequence::FmodLogoFadeOut:
+				if (m_SectionSwitch) { SetSectionDurationAndResetSwitch(0.5F); }
+				m_FadeAmount = static_cast<int>(LERP(0, 1.0F, 0, 255.0F, m_SectionProgress));
+				break;
+			default:
+				break;
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void TitleScreen::UpdateIntroSlideshowSequence(bool skipSlideshow) {
+		if (skipSlideshow && (m_IntroSequenceState > IntroSequence::SlideshowFadeIn && m_IntroSequenceState != IntroSequence::MainMenuAppear)) {
+			m_SectionSwitch = true;
+			m_IntroSequenceState = IntroSequence::MainMenuAppear;
+			m_GameLogo.SetPos(Vector(static_cast<float>(g_FrameMan.GetResX() / 2), 120));
+			m_StationOrbitTimer.SetElapsedRealTimeS(19);
+			return;
+		}
+		m_SlideshowSlideText.clear();
+
+		switch (m_IntroSequenceState) {
+			case IntroSequence::SlideshowFadeIn:
+				if (m_SectionSwitch) {
+					SetSectionDurationAndResetSwitch(1.5F);
+					m_IntroSongTimer.SetElapsedRealTimeS(0.05F);
+					m_IntroScrollStartTime = static_cast<float>(m_IntroSongTimer.GetElapsedRealTimeS());
+					m_IntroScrollDuration = 66.6F - m_IntroScrollStartTime;
+					m_ScrollOffset.SetY(m_IntroScrollStartOffsetY);
+
+					g_AudioMan.PlayMusic("Base.rte/Music/Hubnester/ccintro.ogg", 0);
+					g_AudioMan.SetMusicPosition(0.05F);
+				}
+				m_FadeAmount = static_cast<int>(LERP(0, 1.0F, 255.0F, 0, m_SectionProgress));
+				break;
+			case IntroSequence::PreSlideshowPause:
+				if (m_SectionSwitch) { SetSectionDurationAndResetSwitch(3.2F - static_cast<float>(m_IntroSongTimer.GetElapsedRealTimeS())); }
```

Comment:
This is a real clever way to sync the slides to the song! I'd never noticed that the duration shifts as needed, which is a good sign for how smoothly done it is.